doc/HD44780.rst

```diff
@@ -0,0 +1,197 @@
+HD44780
+-------
+
+.. image:: images/hd44780_clock.jpg
+
+The HD44780 is a popular LCD display that is widely available and cheap.  It
+is available in several sizes with 16 character x 2 line and  20 character
+by 4 line displays the most common.  As with most LCD displays, they come with
+a backlight which Luma.LCD can control either using PWM modulation or with a
+simple GPIO driven switch.
+
+They are normally connected to SBCs using a parallel ('bit bang') style interface
+but are also sold pre-connected to a few different I2C backpacks including the
+PCF8574, MCP23008 and MCP23017 expanders.
+
+.. Note:
+  Currently luma.lcd only supports the parallel and pcf8574 interfaces
+
+
+Capabilities
+^^^^^^^^^^^^
+The HD44780 is mainly a character based device and can be purchased with different
+font tables installed in its ROM.  The two most popular are the A00 ROM (English/Japanese)
+and the A02 ROM (English/European).  See :download:`HD44780 <tech-spec/HD44780.pdf>`
+for details on which characters are included in each.
+
+In addition to the built-in characters, HD44780s can have up to eight custom
+written into their RAM.  This feature is used by Luma.LCD to provide a limited
+graphics capability.
+
+Character-mode Usage
+^^^^^^^^^^^^^^^^^^^^
+When using :py:class:`luma.lcd.device.hd44780` the most common way of controlling
+the display is to use the `text` property which operates similarly to the
+:py:class:`luma.core.virtual.sevensegment` wrapper class.  To use it, assign `text`
+a string containing the values that you want displayed.
+
+.. code:: python
+
+  from luma.core.interface.serial import parallel
+  from luma.lcd.device import hd44780
+
+  p = parallel(RS=7, E=8, PINS=[25,24,23,27])
```

Comment:
no need to use single char vars, use something descriptive instead.

Reply:
They are for the (R)egister(S)elect pin and the (E)nable pin.  Anyone wiring these displays should know what these mean and the naming convention is consistent with the ht1621 class.  Take a look and if you still want them changed, I'll rename them.

Reply:
Was @thijstriemstra talking about `p` (for the parallel interface) and `d` (for the device) ?

As an aside, I'm also wondering if `parallel` is  in the right package (`from luma.core.interface.serial import parallel`) and whether it is a good name for the interface ... perhaps it aught be something like `from luma.core.interface.parallel import bit_bang` ?

Reply:
Ahh, I misunderstood.  I can adjust the variable names in the example.  

To your other point, I can certainly move the parallel class into its own module within luma.lcd but I'm wondering if there is a different way that the interfaces should be organized.  The common interfaces I am aware of are...

* Parallel 6800
* Parallel 8080
* SPI 3 wire
* SPI 4 wire
* I2C

In addition, I2C backpacks such as the PCF8574, MCP23008 and the MCP23017 are often used with the I2C interface to bitbang devices that only support a parallel interface. 

Perhaps collapsing the serial interfaces into the interface module instead of considering them their own interface type.

luma.core.interface
* SPI
* I2C
* PCF8574
* MCP23008
* MCP23017
* ftdi_spi
* ftdi_i2c
* bitbang_6800
* bitbang_8080
* bitbang_spi
* bitbang_serial (for devices like the ht1621)